docs/DatabaseInjector.md

```diff
@@ -0,0 +1,106 @@
+
+```sh
+helm install nuodb-cp-operator nuodb-cp/nuodb-cp-operator \
+    --namespace nuodb-cp-system \
+    --set cpOperator.watchNamespace.enabled=false \
+    --set nuodb.serviceAccount.create=false \
+    --set nuodb.serviceAccount.name=default \
+    ...
+```
+
+>**Note**
+> The `nuodb` service account (SA) creation is disabled in the above command for simplicity. To enable NuoDB Kubernetes Aware Admin (KAA) capabilities, the NuoDB processes should be given special permissions to access Kubernetes API server.
+For more information, please check [Automatic Management of NuoDB State](https://doc.nuodb.com/nuodb/latest/deployment-models/kubernetes-environments/kubernetes-aware-admin/). For cluster-scoped deployments, the NuoDB SA and RBAC should be provisioned before hand in each namespace where NuoDB databases are created.
+
+## Database Injection
+
+NuoDB operator can inject database information into ConfigMap's _data_ once the database is ready.
+This enables easy data source configuration in the application container and acts as a dependency mechanism without the need of additional _init_ containers.
+
+### Creating Database
+
+NuoDB domain and database resources can be created either via REST API or declaratively using [custom resources][4].
+This example creates NuoDB domain and database using custom resources.
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/nuodb/nuodb-cp-releases/main/docs/files/domain.yaml
+kubectl apply -f https://raw.githubusercontent.com/nuodb/nuodb-cp-releases/main/docs/files/database.yaml
+```
+
+## Creating Sample Application
+
+Create a sample Yahoo! Cloud Serving Benchmark (YCSB) application and reference the database information into the _app_ container.
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/nuodb/nuodb-cp-releases/main/docs/files/ycsb-demo-app.yaml
+```
+
+Wait for the database to become ready.
+
+```sh
+kubectl wait --for=condition=ready database acme-messaging-demo
+```
+
+Verify that database information is injected into `acme-messaging-demo-info` ConfigMap's data and the YCSB Pod is _Running_.
+
+```sh
+kubectl get cm acme-messaging-demo-info -o yaml
+kubectl get pods -l app=ycsb-load
+```
+
+### Injected Properties
+
+| Property | Description |
+| ----- | ----------- |
+| `dbName` | NuoDB database name |
+| `dbHost` | The FQDN of the domain managing this database. If external access is enabled on the domain, the property will be populated with the external FQDN. |
+| `dbPort` | The database port for SQL clients. If external access is enabled on the domain, the property will be populated with the Ingress Controller's service port (by default _443_). |
+| `caPem` | The Certificate Authority (CA) certificate for the domain. Used by SQL clients that need to enable TLS encryption on the database connections. If TLS is not enabled on the domain, the property is not injected. |
```

Comment:
Following up on my comment above, it might be good to put that subsection after this one so that it is clearer what it is talking about.